### DISCUSSION TITLE
Safety measures for remote nodes

### DISCUSSION THREAD

**Discussion by SquishyBubble** [meshtastic/Discussions#28] [General]
I've taken down my remote solar-powered nodes for the winter. They've worked well but I want to improve the design to the next season. My greatest concern is fire safety. Below I've listed the planned upgrades to lessen the fire hazard of a remote node, exposed to the elements the better part of the year.

- Temperature switches
    - Normally open, closing at 10 °C - preventing battery charging at low temperatures
    - Normally closed, opening at 45 °C - disconnecting the battery at high temperatures
- Thermal fuse rated at 65 °C - permanently disconnecting a node reaching high temperatures indicative of a fault
- Battery of limited capacity paired with a relatively large solar panel. This makes the maximum energy stored in the node less dangerous.
- Telemetry
  - Temperature and humidity - the measurements can be plotted over time and a node gradually heating up over several days may be taken down before dangerous conditions occur.
  - Voltage/current telemetry. Using an INA3221, a 3-channel chip, lets me monitor both the solar panel, battery and the MCU itself. A server at home can remotely shut down a node operating outside of a chosen envelope.
- Including a second, sand filled inner metal casing housing the battery. The battery will be well isolated from possible metal in the sand.

Have you got any further ideas/comments to improve the fire safety of remote nodes?

**Comment by ikearat** [comment]
Battery in a "detachable" enclosure, or completely remote mounted. The idea is, should enough heat be present, the battery will fall away from the node and not destroy the node. The fall-away can be managed by having the battery fall into the void space in a stack of cinder blocks. Alternately, remote mount the battery in a cinder block packed in sand. You could even bury the block and battery as a crude form of temperature control. Around here, theft would be my first concern. Even the Geocaches get looted here :/

**Reply by adingbatponder** [reply]
this has some suggestions too
https://meshtastic.discourse.group/t/battery-paranoia/15173

**Reply by h0lad** [reply]
Modern battery charge controllers have all safety features integrated. Including High- and Low-temperature shutoff.

They are super easy to implement into a PCB layout.

![image](https://github.com/user-attachments/assets/9f27b5df-a900-4691-bf38-22e2d5042427)

**Reply by h0lad** [reply]
They are fully solar compatible. I implemented a similar one (BQ25185 - a few features less and no I2C) in my own Meshtastic PCB designs  without problems and tested it with 6V/12V/18V panels.

For them a solar panel is just a high impedance source. It automatically adjusts the maximum current draw using VINDPM which keeps it a bit below the collapsing level - which is basically a shabby pseudo-MPPT.

**Reply by adingbatponder** [reply]
Super. Good to know. Do you have the PCB design published ? This one ? https://github.com/dm5tt/MeshtasticNode_BQ25185

**Reply by h0lad** [reply]
Yes. That (https://github.com/dm5tt/MeshtasticNode_BQ25185) one.

**Reply by h0lad** [reply]
But beware: the BQ25185 only has a high-temperature shut-off. No low temperature detection. For that feature you will need the variant mentioned above.

I tested it with a hot air station... worked.

**Comment by adingbatponder** [comment]
This device allows you to charge only when the temperature is ok https://www.adafruit.com/product/4755

**Reply by h0lad** [reply]
That old thing has many restrictions.

With the BQ25186 you can use panels up to 18V which gives the the margin you will need during the winter time is is configurable regarding the high  and low temperature limits.
